File: src/shared/types/docker.ts
export interface Container {
  id: string;
  name: string;
  image: string;
  state: "Running" | "Paused" | "Exited" | "Created" | "Restarting" | "Dead";
  status: string;
  ports: PortMapping[];
  created_at: string;
}

export interface PortMapping {
  host_port: number | null;
  container_port: number;
  protocol: string;
}

export interface DockerImage {
  id: string;
  repository: string;
  tag: string;
  size: number;
  created_at: string;
}

export interface DockerStatus {
  available: boolean;
  containers: Container[];
  images: DockerImage[];
}

File: src-tauri/src/domain/entities/docker.rs
use crate::domain::value_objects::MemorySize;
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Container {
    pub id: String,
    pub name: String,
    pub image: String,
    pub state: ContainerState,
    pub status: String,
    pub ports: Vec<PortMapping>,
    pub created_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum ContainerState {
    Running,
    Paused,
    Exited,
    Created,
    Restarting,
    Dead,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PortMapping {
    pub host_port: Option<u16>,
    pub container_port: u16,
    pub protocol: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DockerImage {
    pub id: String,
    pub repository: String,
    pub tag: String,
    pub size: MemorySize,
    pub created_at: DateTime<Utc>,
}
